Add Vercel middleware docs to Vercel integration README (#5576)
withastro/astro: packages/integrations/vercel/README.md

import vercel from '@astrojs/vercel/static';
```

> ### Middleware

Reviewer: I’d suggest a slightly more descriptive title here. This will show up in search results on our docs site and would be a slightly misleading title in that context if someone was looking for more generic middleware information.

```suggestion
### Vercel Middleware
```

You can use vercel middleware to get the request and redirect before the user loaded the page. Vercel middleware can run on both astro Edge, SSR or Static. You don't need to install `@vercel/edge` to access the middleware but you can install it to get feature such as geolocation. For more information about [middleware](https://vercel.com/docs/concepts/functions/edge-middleware)

> 1. Configure your `astro.config.mjs`
> 
>     ```js
>     // astro.config.mjs
>     import { defineConfig } from 'astro/config';
>     import vercel from '@astrojs/vercel/static';
> 
>     export default defineConfig({
>       output: 'static', // 'server' or 'edge'
>       adapter: vercel(),
>     });
>     ```

Reviewer: There’s nothing unique about this step right? It’s the same configuration you’d use when deploying without middleware? I’d suggest removing it in that case.

```suggestion
```

> 1. Add `middleware.js` to your root of your project. 

Reviewer: ```suggestion
1. Add a `middleware.js` file to the root of your project:
```

    ```js

>     //middleware.js

Reviewer: Missing space here:

```suggestion
    // middleware.js
```

    export const config = {

>     // Only run the middleware on the admin route
>     matcher: '/admin',

Reviewer: One more level of indentation needed here:

```suggestion
      // Only run the middleware on the admin route
      matcher: '/admin',
```

    };

  export default function middleware(request) {
    const url = new URL(request.url);

>     //you can here get ip location or cookie

Reviewer: ```suggestion
    // You can retrieve IP location or cookies here.
```

>     if(url.pathname === "/admin"){

Reviewer: Code style nit:

```suggestion
    if (url.pathname === "/admin") {
```

      url.pathname = "/"
    }
    return Response.redirect(url);
  }
    ```

> 1. Run `vercel dev` to start running middleware. 

Reviewer: This is only on your local machine right? In production Vercel will handle this for you? In that case, I’d probably say so explicitly:

```suggestion
1. While developing locally, you can run `vercel dev` to run middleware. In production, Vercel will handle this for you.
```

});
```

### Vercel Middleware

You can use Vercel middleware to intercept a request and redirect before sending a response. Vercel middleware can run for Edge, SSR, and Static deployments. You don't need to install `@vercel/edge` to write middleware, but you do need to install it to use features such as geolocation. For more information see [Vercel’s middleware documentation](https://vercel.com/docs/concepts/functions/edge-middleware).

1. Add a `middleware.js` file to the root of your project:

    ```js
    // middleware.js
    export const config = {
      // Only run the middleware on the admin route

>       matcher: '/admin',

Reviewer 2: is there a better path we could use here? `/admin` is a peculiar route to be using
